actionview/lib/action_view/template/handlers/erb.rb

```diff
@@ -1,68 +1,65 @@
-require "erubis"
+require "erubi"
 
 module ActionView
   class Template
     module Handlers
-      class Erubis < ::Erubis::Eruby
-        def add_preamble(src)
+      class Erubi < ::Erubi::Engine
+        def initialize(input, properties = {})
           @newline_pending = 0
-          src << "@output_buffer = output_buffer || ActionView::OutputBuffer.new;"
+
+          properties = Hash[properties]
+          properties[:preamble] = "@output_buffer = output_buffer || ActionView::OutputBuffer.new;"
+          properties[:postamble] = "@output_buffer.to_s"
+          properties[:bufvar] = "@output_buffer"
+          properties[:escapefunc] = ""
+          super
         end
 
-        def add_text(src, text)
+        def evaluate(_context)
+          _context.instance_eval(&eval("proc { #{@src} }", binding, @filename || "(erubi)"))
+        end
+
+      private
+
+        def add_text(text)
           return if text.empty?
 
           if text == "\n"
             @newline_pending += 1
           else
             src << "@output_buffer.safe_append='"
             src << "\n" * @newline_pending if @newline_pending > 0
-            src << escape_text(text)
+            src << text.gsub(/['\\]/, '\\\\\&')
```

Comment:
Because having a method that is only caused in one place is slower, and it seems unlikely someone would like to override the behavior.  I'll go ahead and add an escape_text method in the Rails ERB handler.